Source/Scene/ShadowMap.js

```diff
@@ -0,0 +1,397 @@
+
+    /**
+     * @private
+     */
+    function ShadowMap(scene) {
+        this.enabled = true;
+
+        this.debugShow = false;
+        this.debugFreezeFrame = false;
+        this._debugLightFrustum = undefined;
+        this._debugLight = undefined;
+        this._debugCameraFrustum = undefined;
+        this._debugCamera = undefined;
+        this._debugShadowViewCommand = undefined;
+
+        this._depthStencilTexture = undefined;
+        this._framebuffer = undefined;
+        this._size = 1024; // Width and height of the shadow map in pixels
+
+        this.renderState = RenderState.fromCache({
+            cull : {
+                enabled : true,
+                face : CullFace.BACK
+            },
+            depthTest : {
+                enabled : true
+            },
+            colorMask : {
+                red : false,
+                green : false,
+                blue : false,
+                alpha : false
+            },
+            depthMask : true,
+            viewport : new BoundingRectangle(0, 0, this._size, this._size)
+        });
+
+        this.passState = new PassState(scene.context);
+
+        // TODO : determine frustum based on scene
+        var frustum = new OrthographicFrustum();
+        frustum.left = -15.0;
+        frustum.right = 15.0;
+        frustum.bottom = -15.0;
+        frustum.top = 15.0;
+        frustum.near = 1.0;
+        frustum.far = 100.0;
+        this._frustum = frustum;
+
+        // TODO : position camera based on sun direction
+        var camera = new Camera(scene);
+        camera.frustum = frustum;
+        var centerLongitude = -1.31968;
+        var centerLatitude = 0.698874;
+        camera.lookAt(Cartesian3.fromRadians(centerLongitude, centerLatitude), new Cartesian3(0.0, 0.0, 75.0));
+        this.camera = camera;
+
+        this.clearCommand = new ClearCommand({
+            depth : 1.0,
+            renderState : this.renderState,
+            framebuffer : undefined, // Set later
+            owner : this
+        });
+    }
+
+    function destroyTextures(shadowMap) {
+        shadowMap._depthStencilTexture = shadowMap._depthStencilTexture && !shadowMap._depthStencilTexture.isDestroyed() && shadowMap._depthStencilTexture.destroy();
+    }
+
+    function destroyFramebuffers(shadowMap) {
+        shadowMap.framebuffer = shadowMap.framebuffer && !shadowMap.framebuffer.isDestroyed() && shadowMap.framebuffer.destroy();
+    }
+
+    function createTextures(shadowMap, context) {
+        // TODO : Use nearest filtering for testing, not default linear.
+        shadowMap._depthStencilTexture = new Texture({
+            context : context,
+            width : shadowMap._size,
+            height : shadowMap._size,
+            pixelFormat : PixelFormat.DEPTH_STENCIL,
+            pixelDatatype : PixelDatatype.UNSIGNED_INT_24_8
+        });
+    }
+
+    function createFramebuffers(shadowMap, context) {
+        destroyTextures(shadowMap);
```

Comment:
I just mean move the first three function calls in this function to the right above where `createFramebuffers` is called.